web/src/stream_topic_history_util.ts

```diff
@@ -39,3 +40,50 @@ export function get_server_history(stream_id: number, on_success: () => void): v
         },
     });
 }
+
+export function update_topic_last_message_id(
+    stream_id: number,
+    topic_name: string,
+    update_dom_on_success: () => void,
+): void {
+    void channel.get({
+        url: "/json/messages",
+        data: {
+            narrow: JSON.stringify([
+                {operator: "stream", operand: stream_id},
+                {operator: "topic", operand: topic_name},
+            ]),
+            anchor: "newest",
+            num_before: 1,
+            num_after: 0,
+        },
+        success(data) {
+            const {messages} = z
+                .object({
+                    messages: z.array(
+                        z.object({
+                            id: z.number(),
+                        }),
+                    ),
+                })
+                .parse(data);
+            if (messages.length !== 1) {
+                return;
+            }
+
+            const last_message = messages[0];
+            assert(last_message !== undefined);
+            stream_topic_history.add_history(stream_id, [
+                {
+                    name: topic_name,
+                    max_id: last_message.id,
+                },
+            ]);
+            update_dom_on_success();
+        },
+        error() {
+            // Ideally we would retry since we should always be able to get a success response
+            // from the server for this request, but for now we just ignore the error.
+        },
+    });
+}
```

Comment:
This is fine, but I'd prefer to have a `with_last_message_id` function a la `with_first_message_id`; that felt like a nice abstraction of most of the logic.